components/multi-attribute-login/org.wso2.carbon.identity.unique.claim.mgt/src/main/java/org/wso2/carbon/identity/unique/claim/mgt/listener/UniqueClaimUserOperationEventListener.java

```diff
@@ -223,6 +212,58 @@ public boolean isUniqueClaim(String claimUrI, String tenantDomain) throws ClaimM
         return false;
     }
 
+    private void checkUsernameUniqueness(String username, UserStoreManager userStoreManager) throws UserStoreException {
+
+        String errorMessage;
+        String tenantDomain = getTenantDomain(userStoreManager);
+
+        try {
+            if (isUniqueClaim(USERNAME_CLAIM, tenantDomain) &&
+                    isClaimDuplicated(username, USERNAME_CLAIM, username, null, userStoreManager)) {
+
+                errorMessage = "Username " + username + " is already in use by a different user!";
+                throw new UserStoreException(errorMessage, new PolicyViolationException(errorMessage));
+            }
+        } catch (ClaimMetadataException e) {
+            log.error("Error while getting claim metadata for claimUri : " + USERNAME_CLAIM + ".", e);
+        }
+    }
+
+    private String getTenantDomain(UserStoreManager userStoreManager) throws UserStoreException {
+
+        try {
+            return UniqueClaimUserOperationDataHolder.getInstance().getRealmService().getTenantManager().
+                    getDomain(userStoreManager.getTenantId());
+
+        } catch (org.wso2.carbon.user.api.UserStoreException e) {
+            throw new UserStoreException("Error while retrieving tenant domain.", e);
+        }
+    }
+
+    private boolean isScopeWithinUserstore() {
+
+        String scopeWithinUserstore =
+                (String) IdentityUtil.readEventListenerProperty(UserOperationEventListener.class.getName(),
+                        UniqueClaimUserOperationEventListener.class.getName()).getProperties().get(
+                        SCOPE_WITHIN_USERSTORE);
```

Comment:
What is the usecase of SCOPE_WITHIN_USERSTORE? Shall we add a comment

Reply:
Added a comment. This was added with https://github.com/wso2/product-is/issues/12360.